src/V3EmitCBase.h

```diff
@@ -105,6 +127,49 @@ class EmitCBaseVisitor VL_NOT_FINAL : public AstNVisitor {
         return args;
     }
 
+    void emitCFuncHeader(const AstCFunc* funcp, const AstNodeModule* modp, bool withScope) {
+        if (!funcp->isConstructor() && !funcp->isDestructor()) {
+            puts(funcp->rtnTypeVoid());
+            puts(" ");
+        }
+        if (withScope && funcp->isProperMethod()) puts(prefixNameProtect(modp) + "::");
+        puts(funcNameProtect(funcp));
+        puts("(" + cFuncArgs(funcp) + ")");
+        if (funcp->isConst().trueKnown() && funcp->isProperMethod()) puts(" const");
+    }
+
+    void emitCFuncDecl(const AstCFunc* funcp, const AstNodeModule* modp) {
+        ensureNewLine();
+        if (!funcp->ifdef().empty()) puts("#ifdef " + funcp->ifdef() + "\n");
+        if (funcp->isStatic().trueUnknown() && funcp->isProperMethod()) puts("static ");
+        if (funcp->isVirtual()) {
+            UASSERT_OBJ(funcp->isProperMethod(), funcp, "Virtual function is not a proper method");
+            puts("virtual ");
+        }
+        emitCFuncHeader(funcp, modp, /* withScope: */ false);
+        if (funcp->slow()) puts(" VL_ATTR_COLD");
+        puts(";\n");
+        if (!funcp->ifdef().empty()) puts("#endif  // " + funcp->ifdef() + "\n");
+    }
+
+private:
+    static bool isWordChar(char c) { return isalnum(c) || c == '_'; }
+
+public:
+    static string replaceWord(string str, const string& from, const string& to) {
+        const size_t len = from.size();
+        UASSERT_STATIC(len > 0, "Cannot replace empty string");
+        for (size_t pos = 0; (pos = str.find(from, pos)) != string::npos; pos += len) {
+            // Only replace whole words
+            if (((pos > 0) && isWordChar(str[pos - 1])) ||  //
+                ((pos + len < str.size()) && isWordChar(str[pos + len]))) {
+                continue;
+            }
+            str.replace(pos, len, to);
+        }
+        return str;
+    }
```

Comment:
Perhaps better in V3String.  Also if len is zero one would expect it to just return "" as there's no replacements to do.

Reply:
Moved to V3String. The assert is over the length of the substring to replace, i.e.: this blows up: `replaceWord("banana", "", "HUH??")`